iml-gui/crate/src/components/command_modal.rs

```diff
@@ -284,3 +548,638 @@ fn close_button() -> Node<Msg> {
     ]
     .map_msg(Msg::Modal)
 }
+
+async fn fetch_the_batch<T, F, U>(ids: Vec<u32>, data_to_msg: F) -> Result<U, U>
+where
+    T: DeserializeOwned + EndpointName + 'static,
+    F: FnOnce(ResponseDataResult<ApiList<T>>) -> U,
+    U: 'static,
+{
+    // e.g. GET /api/something/?id__in=1&id__in=2&id__in=11&limit=0
+    let err_msg = format!("Bad query for {}: {:?}", T::endpoint_name(), ids);
```

Comment:
It is executed unconditionally, even when the request is succeeding (in most cases, I hope).